tests/sklearn/test_sklearn_autolog.py

```diff
@@ -621,7 +698,7 @@ def test_parameter_search_handles_large_volume_of_metric_outputs():
     with mlflow.start_run() as run:
         svc = sklearn.svm.SVC()
         cv_model = sklearn.model_selection.GridSearchCV(
-            svc, {"C": [1]}, n_jobs=1, scoring=metrics_to_log, refit=False
+            svc, {"C": [1]}, n_jobs=1, scoring=metrics_to_log, refit="score_0"
```

Comment:
I made this modification in order to incorporate my added unit tests.
I found that **GridSearchCV**'s base class is **Classifier**, to it will execute the classifier metrics logging, where I will always use the passed in estimator as an already trained model and call **predict()**
If you set refit=False, then the estimator will not refit itself according to the metric you put. So here I just arbitrarily pick **refit=score_0** so that the model can get refit.

Another way to solve this case is we can guard the **GridSearchCV** from going into the code path for metric logging, but this will be based on whether we want to log common metrics for **GridSearchCV** or not.